src/index.ts

```diff
@@ -245,9 +246,12 @@ export async function runInContext(
       )
     }
   } else {
-    const it = evaluate(program, context)
+    let it = evaluate(program, context)
     let scheduler: Scheduler
-    if (theOptions.scheduler === 'async') {
+    if (context.executionMethod === 'non-det-interpreter') {
```

Comment:
As discussed earlier, we can choose to either let the IOOptions know the scheduler type, or let the context know the execution method

Reply:
We'd still still need to let the context know the execution method as otherwise it will be run as native code